tinygrad/tensor.py

```diff
@@ -67,12 +68,16 @@ class Tensor:
   __slots__ = "lazydata", "requires_grad", "grad", "_ctx"
   __deletable__ = ('_ctx',)
   training: ClassVar[bool] = False
-  class train:
-    def __init__(self, val=True): self.val = val
-    def __enter__(self): self.prev, Tensor.training = Tensor.training, self.val
+  class train(ContextDecorator):
+    def __init__(self, mode:bool = True): self.mode = mode
```

Comment:
Err, why did this one change? Does this match torch?

Reply:
Yes, torch also uses `mode` for `train`. See [source code](https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module)